dashboard/test/models/user_test.rb

```diff
@@ -5473,6 +5473,47 @@ def hide_scripts_in_sections(section1, section2)
     assert_equal new_us_state, student.reload.us_state
   end
 
+  test 'teacher in password strict country requires longer password length' do
+    user = User.new(
+      user_type: User::TYPE_TEACHER,
+      country_code: User::PASSWORD_STRICT_COUNTRIES.first,
+      password: 'short',
+      password_confirmation: 'short'
+    )
+    DCDO.stubs(:get).with('strict-password-country', false).returns(true)
+
+    refute user.valid?
+    assert_includes user.errors[:password], 'is too short (minimum is 14 characters)'
+
+    user = User.new(
+      user_type: User::TYPE_TEACHER,
+      country_code: User::PASSWORD_STRICT_COUNTRIES.first,
+      password: 'longlongpassword',
+      password_confirmation: 'longlongpassword'
+    )
+
+    user.valid?
+    assert_empty user.errors[:password]
```

Comment:
Keep each assertion in a separate test, so that if one of the tests fails, it's clearer if the test failing is the `short` or `longpassword`.